src/main/java/software/amazon/smithy/gradle/SmithyPlugin.java

```diff
@@ -94,9 +99,39 @@ public void apply(Project project) {
         task.setDuplicatesStrategy(DuplicatesStrategy.EXCLUDE);
         task.dependsOn(buildJarProvider);
 
+        Jar jar = project.getTasks().withType(Jar.class).getByName("jar");
+//        jar.doFirst(unused -> {
+//            addSmithyTags(jar, project, extension);
+//        });
+
+//        jar.doFirst(new Action<Task>() {
+//            @Override
+//            public void execute(Task task) {
+//                addSmithyTags(jar, project, extension);
+//            }
+//        });
+
+        jar.doFirst(new SmithyTagsAction());
```

Comment:
These are 3 different approaches.
A) Java Lamba
This showed up in log:
`  Additional action for task ':jar': was implemented by the Java lambda 'software.amazon.smithy.gradle.SmithyPlugin$$Lambda$5794/537906986'. Using Java lambdas is not supported, use an (anonymous) inner class instead.`

B) Anonymous Inner class
To avoid the log line from (A).

C) Similar to (B), but separate SmithyTagsAction class. I thought it might have an advantage of `@Input` to deal with changing tags in build.gradle, but it didn't help. Also has a limitation that SmithyTagsAction is defined with `Action<Task>`, though `Action<Jar>` would have been ideal, but `doFirst`'s signature is `Action<? super Task>` so had to do `Action<Task>` and cast to `Jar` in `execute` - so not ideal.